src/compaction.c

```diff
@@ -88,6 +94,47 @@ void AvgReadContext(void *contextPtr, RedisModuleIO * io){
     context->cnt = RedisModule_LoadDouble(io);
 }
 
+void *StdCreateContext() {
+    StdContext *context = (StdContext *)malloc(sizeof(StdContext));
+    context->cnt = 0;
+    context->mean =0;
+    context->std =0;
+    return context;
+}
+
+void StdAddValue(void *contextPtr, double value){
+    StdContext *context = (StdContext *)contextPtr;
+    double oldMean = context->mean;
+    context->mean = oldMean + (value - oldMean) / ++context->cnt;
```

Comment:
you can't caculate the mean this way. you are changing the weight of the samples

Reply:
There are an online algorithms for standard deviation. Population Variance was  implemented. Sample variance can be added.

Reply:
If would replace this one std with 2 or maybe 4 commands. 
'stdp' and 'stds' for either population or samples std. Then maybe also 'varp' and 'vars' (or maybe with a dot in between) which is the square root of the std. The only difference between these 4 is whether it is divided by 'N' or 'N - 1' and whether we sqrt() the result.